common/network-common/src/main/java/org/apache/spark/network/util/TransportFrameDecoder.java

```diff
@@ -139,14 +139,18 @@ private ByteBuf decodeNext() throws Exception {
       return nextBufferForFrame(remaining);
     }
 
-    // Otherwise, create a composite buffer.
-    CompositeByteBuf frame = buffers.getFirst().alloc().compositeBuffer();
```

Comment:
Why is it not a good choice? With your change, you're replacing "maybe copy multiple times" with "always copy once". If there's a way to avoid the copy altogether, why not do it?